enhancements/sdk-supports-operands-deletion.md

```diff
@@ -0,0 +1,118 @@
+
+However, cross-namespaced owner references are not allowed by design and this is where the operator would need to leverage using [Finalizers][finalizers]. to perform cleanup of operands by processing certain deletion events.
+
+SDK will encourage and enable the use of finalizers when it is required to. By default, all operators would have scaffolded logic for Finalizers.
+
+## Goals
+
+- Provide default scaffolding for an operand cleanup
+- Cleanup is strictly on an opt-in basis only and not automatic by default
+- Cleanup is limited to removal of CRs and not CRDs
+- SDK will have additional flags to prevent deletion of CRDs and instead use OLM's operand cleanup feature
+
+## Non-Goals
+
+- SDK support in handling cleaning up resources other than CRs  
+
+## Possible Future Goals (Not in scope)
+
+- SDK should have a mechanism to handle events/alerts received by OLM in case cleanup is blocked
+- The timeout for operand deletion should be user configurable by the SDK
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want an upstream finalizer no-op handler scaffolded by default and/or library to help write finalizers.
+
+#### Story 2
+
+As an operator author, I want scaffolded fields in the CSV generation by default to enable operand cleanup.
+
+#### Story 3
+
+As an operator author, I would like the Operator SDK's cleanup command to handle deletion of CRDs appropriately based on the flags provided.
+
+### Implementation Details/Notes/Constraints
+
+#### Upstream finalizer no-op handler scaffolded by default and/or library to help write Finalizers
+
+Any operator that handles delete events or if the operator is expecting a delete event for CRs should have Finalizers. The only time Finalizers may not be required is when OwnerReferences exist and they handle all of the cleanup. But as we tend towards cluster scoped operators, OwnerReferences won't be allowed. In that case, all operators should have Finalizers.
+
+To enable this, it would be ideal to have a Finalizer no-op handler scaffolded by default in upstream such as Kubebuilder. The idea is to always scaffold out a Finalizer, a Delete Event Handler and a Reconciler that checks for Finalizers and has a no-op in cases where it's not required to use a Finalizer. 
+
+The Reconciler checks for presence of a Finalizer in a CR object. If the CR object is marked for deletion, which is indicated by the timestamp that was set, the Reconciler runs the finilization logic and removes the finalizer. If the finalization logic fails for some reason, then the Reconciler keeps the Finalizer to retry during the next reconcilation. Typically, the finalization logic would entail necessary cleanup steps that the operator needs to perform before the CR can be deleetd.
+
+Note: If a Finalizer doesn't exist on the CR and the object is marked for delete event, the Reconciler would add a Finalizer to the CR object as part of reconcilcation.
```

Comment:
Why would the operator do this? Shouldn't the finalizer be added to all CRs prior to deletion, so that full deletion doesn't occur until the finalizer is removed?